include/range/v3/view/filter.hpp

```diff
@@ -97,6 +97,35 @@ namespace ranges
             }
         };
 
+        /// # ranges::views::filter
+        /// The filter view takes in a predicate function `T -> bool` and converts an
+        /// input range of `T` into an output range of `T` by keeping all elements for
+        /// which the predicate returns true.
+        ///
+        /// ## Example
+        /// \snippet example/view/filter.cpp filter example
+        ///
+        /// ### Output
+        /// \include example/view/filter_golden.txt
+        ///
+        /// ## Syntax
+        /// ```cpp
+        /// auto output_range = input_range | ranges::views::filter(filter_func);
+        /// ```
+        ///
+        /// ## Parameters
+        /// <pre><b>filter_func</b></pre>
+        ///   - Called once for each element of the input range
+        ///   - Returns true for elements that should present in the output range
+        ///
+        /// <pre><b>input_range</b></pre>
+        ///   - The range of elements to filter
+        ///   - Reference type: `T`
+        ///
+        /// <pre><b>output_range</b></pre>
+        ///   - The range of filtered values
```

Comment:
Might be nice to document the the concept satisfied by the resulting range. It is the stronger of either `forward_range` or the concept satisfied by the input. Also, the result doesn't satisfy `sized_range`, and the result is a `common_range` if and only if the input is a `common_range`. The result ranges also does not satisfy `borrowed_range`. I think I got all that right, but it's best to double-check.

Reply:
Added that info.